### PR TITLE
[Utils refactor] new rect package

### PR DESCRIPTION
**Pull request by benoitgrelard**

### CODE REVIEW

File: packages/react/context-menu/src/ContextMenu.tsx [hunk 1]
```diff
@@ -5,14 +5,17 @@ import { makeRect, getSelector } from '@radix-ui/utils';
...
+import type { Point } from '@radix-ui/utils';
 
 /* -------------------------------------------------------------------------------------------------
  * ContextMenu
  * -----------------------------------------------------------------------------------------------*/
 
 const CONTEXT_MENU_NAME = 'ContextMenu';
 
+type ExtractRef<T> = T extends React.RefObject<infer B> ? B : never;
+type MeasurableElement = ExtractRef<React.ComponentProps<typeof MenuPrimitive.Root>['anchorRef']>;
```
Comment (benoitgrelard) [inline packages/react/context-menu/src/ContextMenu.tsx:17]: Is this weird? Should I name `MeasurableElement` something like `Anchor` here instead?

Comment (benoitgrelard) [reply packages/react/context-menu/src/ContextMenu.tsx:17]: My thought process was basically that rather to add an explicit dependency on `@radix-ui/rect` just to get the `MeasurableElement` type, it's prob better to get it from `Menu`, and also carries more intent.

Comment (jjenzz) [reply packages/react/context-menu/src/ContextMenu.tsx:17]: Yeah perhaps just `AnchorElement`?

**Review by jjenzz — APPROVED** [review]

File: packages/react/context-menu/src/ContextMenu.tsx (hunk 1 above)
Comment (benoitgrelard) [reply packages/react/context-menu/src/ContextMenu.tsx]: I went with `Anchor` and renamed the `MeasurableElement` type to `Measurable` as the whole point is that it's not always an element.

**Pull request merged**